File: frontend/src/app/product.service.ts
import { Injectable } from '@angular/core';
import {WebRequestService} from './web-request.service';

@Injectable({
  providedIn: 'root'
})
export class ProductService {

  constructor(private webRequestService:WebRequestService) { }

  createProduct(category:String,name:String,length:number,height:number,weight:number,price:number,numberInStock:number,location:String){
    return this.webRequestService.post('products/newProduct',{category,name,length,height,weight,price,numberInStock,location});
  }
  getProductList(){
    return this.webRequestService.get("products/all");
  }
  getRestockHistory(){
    return this.webRequestService.get("restockhistory/all");
  }
  restockProduct(productID:number,employeeID:number,amountRestocked:number){
   let urlString:string='products/restock/'+productID;
    return this.webRequestService.put(urlString,{employeeID,amountRestocked});
  }
  isEmptyOrWhiteSpaces(text:String){
    return text === null || text.match(/^ *$/) !== null;
  }
  searchProductList(searchText:String){
    if(this.isEmptyOrWhiteSpaces(searchText)){
      return this.getProductList();
    }
    else{
      return this.webRequestService.get("products/search/"+searchText);
    }
  }
  getProductById(searchId:number){
    return this.webRequestService.get(`products/product/${searchId}`);
  }
}

File: src/main/java/com/warehouse/services/ProductService.java
package com.warehouse.services;

import com.warehouse.model.Product;
import com.warehouse.repository.ProductRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Map;


public interface ProductService {


    public List<Product> getAllProducts();

    public Product registerNewProduct(String category,String name,double length, double weight,double height, double price,int numberinstock, String location, int timesshipped);

    public Product registerNewProduct(Product product);

    public List<Product> searchForProduct(String searchText);

    public Product findProductByID(int searchId);
    public Product restockProduct(Map<String,Object> updatedProduct, int productID);
    public Product shipProduct(Map<String,Object> updatedProduct,String id);
    public boolean shipProduct(int amount,int productID);


}
